torch/distributed/tensor/_dtensor_spec.py

```diff
@@ -538,6 +564,12 @@ def device_mesh(self) -> DeviceMesh:
         # checks that mixes DTensor/DTensorSpec easier
         return self.mesh
 
+    @property
+    def shard_order(self) -> Optional[ShardOrder]:
```

Comment:
i'm wondering why this returns an optional shard order.  If we can't convert StridedShard to shard, then should we define shard_order as left to right and return that shard order, to be interpreted alongside our placement tuple which includes stridedshard?